### PR TITLE
Some review comments

### PR DESCRIPTION
**Pull request by agrieve**

### CODE REVIEW

File: www/SyncManager.js [hunk 1]
```diff
@@ -52,15 +53,8 @@ var resolveRegistrations = function(statusVars) {
...
-	options.id = "k" + (new Date()).getTime();
-    } else {
-	options.id = toClone.id;
-    }
-    if (toClone.minDelay != null) {
-	options.minDelay = toClone.minDelay;
-    }
+    options.id = toClone.id || ("k" + idCounter++);
+    options.minDelay = toClone.minDelay || options.minDelay;
     if (toClone.maxDelay != null) {
```
Comment (agrieve) [inline www/SyncManager.js]: Should change the rest as well. Most props will be undefined rather than null I think.

File: www/SyncManager.js [hunk 2]
```diff
@@ -132,16 +121,16 @@ SyncManager.prototype.getRegistrations = function() {
...
+	    resolve(null);  // reject?
 	}
 	exec(innerSuccess, innerFail, "BackgroundSync", "getRegistrations", []);
     });
 }
 
-navigator.serviceWorker.ready.then(function(serviceWorkerRegistration) {
+serviceWorker.ready.then(function(serviceWorkerRegistration) {
     serviceWorkerRegistration.syncManager = new SyncManager();
     exec(syncCheck, null, "BackgroundSync", "initBackgroundSync", []);
```
Comment (agrieve) [inline www/SyncManager.js]: combine this with getRegistrations as well?

File: www/SyncManager.js [hunk 3]
```diff
@@ -52,15 +53,8 @@ var resolveRegistrations = function(statusVars) {
...
-    if (toClone.id == null) {
-	// Use Timestamp as unique id
-	options.id = "k" + (new Date()).getTime();
-    } else {
-	options.id = toClone.id;
-    }
-    if (toClone.minDelay != null) {
-	options.minDelay = toClone.minDelay;
-    }
+    options.id = toClone.id || ("k" + idCounter++);
```
Comment (agrieve) [inline www/SyncManager.js]: changed to a time-seeded counter so as to avoid conflicts on fast devices.

File: www/SyncManager.js [hunk 4]
```diff
@@ -132,16 +121,16 @@ SyncManager.prototype.getRegistrations = function() {
...
 	var innerFail = function(regs) {
-	    resolve(null);
+	    resolve(null);  // reject?
 	}
 	exec(innerSuccess, innerFail, "BackgroundSync", "getRegistrations", []);
     });
 }
 
-navigator.serviceWorker.ready.then(function(serviceWorkerRegistration) {
+serviceWorker.ready.then(function(serviceWorkerRegistration) {
```
Comment (imintz) [inline www/SyncManager.js]: This gives an error since there is no window.serviceWorker object.

**Pull request closed without merging**